### PR TITLE
improve watermark docs: tips of generated columns

### PR DESCRIPTION
**Pull request by fuyufjh**
## Info

- **Description**

Today I was queried about how to use a nested field as a watermark column. Added description and examples here.

- **Notes**

  - [ Include any supplementary context or references here. ]

- **Related code PR**

  - [ Provide a link to the relevant code PR here, if applicable. ]

- **Related doc issue**
  
  Resolves [ Provide a link to the relevant doc issue here, if applicable. ]

## For reviewers

- **Preview**

  - [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **dev** version of the documentation. ]

- **Key points**

  - [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`emile-00`, `hengm3467`, & `WanYixian`).

### CODE REVIEW

**Review by stdrc — APPROVED** [review]
LGTM

File: docs/transform/watermarks.md [hunk 1]
```diff
@@ -59,9 +59,16 @@ For example, the watermark generation strategy can be specified as:
 
     Supported `time_unit` values include: second, minute, hour, day, month, and year. For more details, see the `interval` data type under [Overview of data types](/sql/sql-data-types.md).
 
+
+:::note
+
+Currently, RisingWave only supports using one of the columns from the table as the watermark column. To use nested fields e.g. fields in `STRUCT`, or perform expression evaluation on the input rows e.g. casting data types, please refer to [generated columns](/sql/query-syntax/query-syntax-generated-columns.md).
```
Comment (WanYixian) [inline docs/transform/watermarks.md]:
```suggestion
Currently, RisingWave only supports using one of the columns from the table as the watermark column. To use nested fields (e.g., fields in `STRUCT`), or perform expression evaluation on the input rows (e.g., casting data types), please refer to [generated columns](/sql/query-syntax/query-syntax-generated-columns.md).
```

File: versioned_docs/version-1.9/transform/watermarks.md [hunk 2]
```diff
@@ -59,9 +59,16 @@ For example, the watermark generation strategy can be specified as:
 
     Supported `time_unit` values include: second, minute, hour, day, month, and year. For more details, see the `interval` data type under [Overview of data types](/sql/sql-data-types.md).
 
+
+:::note
+
+Currently, RisingWave only supports using one of the columns from the table as the watermark column. To use nested fields e.g. fields in `STRUCT`, or perform expression evaluation on the input rows e.g. casting data types, please refer to [generated columns](/sql/query-syntax/query-syntax-generated-columns.md).
```
Comment (WanYixian) [inline versioned_docs/version-1.9/transform/watermarks.md]:
```suggestion
Currently, RisingWave only supports using one of the columns from the table as the watermark column. To use nested fields (e.g., fields in `STRUCT`), or perform expression evaluation on the input rows (e.g., casting data types), please refer to [generated columns](/sql/query-syntax/query-syntax-generated-columns.md).
```

**Review by WanYixian — APPROVED** [review]
Others LGTM!

**Pull request merged**